src/components/TopBanner.tsx

```diff
@@ -0,0 +1,38 @@
+import React, { useContext } from 'react'
+import style from "./TopBanner.module.scss"
+import cs from "classnames"
+import { SettingsContext, ISettingsContext } from '../context/Theme'
+import { Warning } from './Layout/Warning';
+
+interface Props {
+ message: string|undefined 
+}
+
+export function TopBanner({ message }:Props) {
+  const settings = useContext<ISettingsContext>(SettingsContext)
+
+  const isOpen = !settings.topBannerIsHidden; 
+
+  const handleClose = () => {
+    if (isOpen && message) {
+      settings.update('topBannerMessageHash', window.btoa(message))
+    }
+  }
+
+  return (
+    <Warning
+      closeButton={
+	<button
+	  className={cs(style.btn_close)}
+	  onClick={handleClose}
+	>
+	  <i aria-hidden className="far fa-times"/>
+	</button>
+      }
+      className={cs({[style.hidden]: !isOpen})}
```

Comment:
The default state should be that the banner is hidden. Because there's a small delay to fetch the local storage, the initial state may not be the one finally computed by the layout once the settings are loaded.
We'd prefer to see a message pop at the top rather to see a weird layout shift to the top when the banner message disappears.
With the current implementation, if we reload the page we may see it shift top and we don't understand why. If we invert this behavior, it makes more sense to see the page shifting down if a message appears at the top.

Reply:
The default value in the store was "hidden". Thats how I wanted to avoid this glitch. anyway's i've updated the implementation: I removed the hidden state from the context because we can store only the message and compute the hidden state from that message within the `TopBanner` itself.